sw/device/tests/dif/dif_otbn_unittest.cc

```diff
@@ -332,32 +332,32 @@ TEST_F(GetErrCodeTest, NullArgs) {
   result = dif_otbn_get_err_code(&dif_otbn_, nullptr);
   EXPECT_EQ(result, kDifOtbnBadArg);
 
-  err_code = kDifOtbnErrorCodeNoError;
+  err_code = kDifOtbnErrCodeBadDataAddr;
```

Comment:
Are you sure you want to perform this assignment? This potentially makes the test flake with false negatives, and is otherwise unnecessary. The following is an antipattern:
```cpp
MyType foo = kWhatever;
function_under_test(&foo);
EXPECT_EQ(foo, kWhatever);
```
If `function_under_test` never sets `foo`, the test still passes. Instead, you should write
```cpp
MyType foo;  // Uninitialized memory.
function_under_test(&foo);
EXPECT_EQ(foo, kWhatever);
```
This style has the advantage that, if we start running unit tests under ASAN or MSAN (which we totally should!), it will catch the "failed to write to `foo`" case.

Reply:
This test checks that the err_code is *not* written; we hence need to assign some known value to it.

Reply:
How do you know that `foo` isn't getting overwritten with the value you initialized it with? In that case, it might make sense to assign an out-of-range value, like all-ones.